packages/@aws-cdk/toolkit/lib/actions/deploy.ts

```diff
@@ -0,0 +1,228 @@
+  /**
+   * Use stack APIs to the deploy stack changes
+   */
+  readonly method: 'direct';
+}
+
+export interface ChangeSetDeploymentMethod {
+  /**
+   * Use change-set APIS to deploy a stack changes
+   */
+  readonly method: 'change-set';
+
+  /**
+   * Whether to execute the changeset or leave it in review.
+   *
+   * @default true
+   */
+  readonly execute?: boolean;
+
+  /**
+   * Optional name to use for the CloudFormation change set.
+   * If not provided, a name will be generated automatically.
+   */
+  readonly changeSetName?: string;
+}
+
+/**
+ * When to build assets
+ */
+export enum AssetBuildTime {
+  /**
+   * Build all assets before deploying the first stack
+   *
+   * This is intended for expensive Docker image builds; so that if the Docker image build
+   * fails, no stacks are unnecessarily deployed (with the attendant wait time).
+   */
+  ALL_BEFORE_DEPLOY,
+
+  /**
+   * Build assets just-in-time, before publishing
+   */
+  JUST_IN_TIME,
+}
+
+export interface Tag {
+  readonly Key: string;
+  readonly Value: string;
+}
+
+export enum RequireApproval {
+  Never = 'never',
+  AnyChange = 'any-change',
+  Broadening = 'broadening',
+}
+
+export enum HotswapMode {
+  /**
+   * Will fall back to CloudFormation when a non-hotswappable change is detected
+   */
+  FALL_BACK = 'fall-back',
+
+  /**
+   * Will not fall back to CloudFormation when a non-hotswappable change is detected
+   */
+  HOTSWAP_ONLY = 'hotswap-only',
+
+  /**
+   * Will not attempt to hotswap anything and instead go straight to CloudFormation
+   */
+  FULL_DEPLOYMENT = 'full-deployment',
+}
+
+export class StackParameters {
+  /**
+   * Use only existing parameters on the stack.
+   */
+  public static onlyExisting() {
+    return new StackParameters({}, true);
+  };
+
+  /**
+   * Use exactly these parameters and remove any other existing parameters from the stack.
+   */
+  public static exactly(params: { [name: string]: string | undefined }) {
+    return new StackParameters(params, false);
+  };
+
+  /**
+   * Define additional parameters for the stack, while keeping existing parameters for unspecified values.
+   */
+  public static withExisting(params: { [name: string]: string | undefined }) {
+    return new StackParameters(params, true);
+  };
+
+  public readonly parameters: Map<string, string | undefined>;
+  public readonly keepExistingParameters: boolean;
+
+  private constructor(params: { [name: string]: string | undefined }, usePreviousParameters = true) {
+    this.keepExistingParameters = usePreviousParameters;
+    this.parameters = new Map(Object.entries(params));
+  }
+}
+
+export interface BaseDeployOptions {
+  /**
+   * Criteria for selecting stacks to deploy
+   */
+  readonly stacks: StackSelector;
+
+  /**
+   * Name of the toolkit stack to use/deploy
+   *
+   * @default CDKToolkit
+   */
+  readonly toolkitStackName?: string;
+
+  /**
+   * Role to pass to CloudFormation for deployment
+   */
+  readonly roleArn?: string;
```

Comment:
`@default`?

Reply:
there's a couple of these elsewhere and i feel like some of this was copy/paste so i'll leave it to you to decide if its necessary. but imo we should be enforcing `@default` tag exists on any optional property, and we were enforcing that at some point, but maybe not in the CLI?